airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/concurrent/adapters.py

```diff
@@ -231,7 +231,12 @@ def read(self) -> Iterable[Record]:
                     yield Record(data_to_return, self)
                 elif isinstance(record_data, AirbyteMessage) and record_data.type == Type.RECORD and record_data.record is not None:
                     # `AirbyteMessage`s of type `Record` should also be yielded so they are enqueued
-                    yield Record(record_data.record.data, self)
+                    # If stream is flagged for file_transfer the record should data in file key
+                    record_message_data = record_data.record.file if self._use_file_transfer() else record_data.record.data
+                    if record_message_data is None:
+                        raise ExceptionWithDisplayMessage("A record without data was found")
```

Comment:
I think this shouldn't happen unless somebody knows a scenario for empty record data "{}", I'm trying to catch any error if the wrong data field is selected/set.